test/TimeService.Data.UnitTests/WorkTimeMonthLimitRepositoryTests.cs

```diff
@@ -0,0 +1,278 @@
+    {
+      Assert.AreEqual(_firstWorkTimeMonthLimit.Id, await _repository.CreateAsync(_firstWorkTimeMonthLimit));
+      Assert.AreEqual(1, await _dbContext.WorkTimeMonthLimits.CountAsync());
+    }
+
+    [Test]
+    public async Task CreateReturnNullAsync()
+    {
+      Assert.IsNull(await _repository.CreateAsync(null));
+      Assert.AreEqual(0, await _dbContext.WorkTimeMonthLimits.CountAsync());
+
+    }
+
+    [Test]
+    public async Task CreateRangeSuccessfullyAsync()
+    {
+      List<DbWorkTimeMonthLimit> dbWorkTimeMonthLimits = new()
+      {
+        _firstWorkTimeMonthLimit, _secondWorkTimeMonthLimit, _thirdWorkTimeMonthLimit
+      };
+
+      await _repository.CreateRangeAsync(dbWorkTimeMonthLimits);
+
+      Assert.AreEqual(3, await _dbContext.WorkTimeMonthLimits.CountAsync());
+    }
+
+    [Test]
+    public async Task CreateRangeReturnNullAsync()
+    {
+      await _repository.CreateRangeAsync(null);
+
+      Assert.AreEqual(0, await _dbContext.WorkTimeMonthLimits.CountAsync());
+    }
+
+    [Test]
+    public async Task CreateRangeReturnNullByOneElementAsync()
+    {
+      List<DbWorkTimeMonthLimit> dbWorkTimeMonthLimits = new()
+      {
+        _firstWorkTimeMonthLimit, null, _thirdWorkTimeMonthLimit
+      };
+
+      await _repository.CreateRangeAsync(dbWorkTimeMonthLimits);
+
+      Assert.AreEqual(0, await _dbContext.WorkTimeMonthLimits.CountAsync());
+    }
+
+    [Test]
+    public async Task EditSuccessfullyAsync()
+    {
+      List<DbWorkTimeMonthLimit> list = new() { _editableWorkTimeMonthLimit };
+      IQueryable<DbWorkTimeMonthLimit> queryable = list.AsQueryable();
+      Mock<DbSet<DbWorkTimeMonthLimit>> dbSet = new Mock<DbSet<DbWorkTimeMonthLimit>>();
+
+      dbSet.As<IQueryable<DbWorkTimeMonthLimit>>()
+        .Setup(x => x.Provider)
+        .Returns(queryable.Provider);
+      dbSet.As<IQueryable<DbWorkTimeMonthLimit>>()
+        .Setup(m => m.Expression)
+        .Returns(queryable.Expression);
+      dbSet
+        .As<IQueryable<DbWorkTimeMonthLimit>>()
+        .Setup(m => m.ElementType)
+        .Returns(queryable.ElementType);
+      dbSet
+        .As<IQueryable<DbWorkTimeMonthLimit>>()
+        .Setup(m => m.GetEnumerator())
+        .Returns(() => queryable.GetEnumerator());
+
+      _mocker
+        .Setup<IHttpContextAccessor, IDictionary<object, object>>(x => x.HttpContext.Items)
+        .Returns(_items);
+      _mocker
+        .Setup<IDataProvider, IEnumerable<DbWorkTimeMonthLimit>>(x => x.WorkTimeMonthLimits)
+        .Returns(dbSet.Object);
+
+      WorkTimeMonthLimitRepository mockedRepository = _mocker.CreateInstance<WorkTimeMonthLimitRepository>();
+
+      string value = "Edited Holidays";
+      JsonPatchDocument<DbWorkTimeMonthLimit> request = new()
+      {
+        Operations = { new() { op = "replace", path = "/Holidays", value = value } }
+      };
+
+      Assert.IsTrue(await mockedRepository.EditAsync(_editableWorkTimeMonthLimit.Id, request));
```

Comment:
Why should not mock class, which we tested.

Reply:
The problem is that I need to mock httpcontextaccessor